pkg/agent/config/config.go

```diff
@@ -80,6 +80,7 @@ func getNodeNamedCrt(nodeName, nodePasswordFile string) HTTPRequester {
 			return nil, err
 		}
 		req.Header.Set(version.Program+"-Node-Password", nodePassword)
+		req.Header.Set("X-K3S-NODE-IP", nodeIP)
```

Comment:
Can you use version.Program.  But keep it "x-" prefixed.  Kind of a shame the node password isn't "x-".

Reply:
I can make a separate pr to update the others if that would be of value.

Reply:
I think the current recommendation is not to use the `X-` prefix https://tools.ietf.org/html/rfc6648

Reply:
I am with @erikwilson on dropping the X- since it is no longer recommended. Can we also match the Title Case style of the node password header? Header keys are technically case insensitive so we might as well make it look nice.